### PR TITLE
Enhance API connection

### LINKED ISSUE #13: Unable to use operator with HTTPS instances
Due to the way the grafana module configures the connection, we're unable to use anything other than plain HTTP Grafana instances.

However, the GrafanaFace connection allows for connection tweaking.

https://github.com/m0nhawk/grafana_api/blob/main/grafana_api/grafana_face.py#L30

### PR DESCRIPTION
**Pull request by unitygilles**
In order to connect the operator to HTTPS instances, we need to augment the current connection setup as part of the `grafana` module.

If environment variable is not set, it defaults to `None` which in turn uses the grafana_api defaults.
https://github.com/m0nhawk/grafana_api/blob/main/grafana_api/grafana_face.py#L20

Closes #13

### CODE REVIEW

**Comment by angelbarrera92** [comment]
Ciao @unitygilles! Thanks for this awesome contribution!

Let me review it :)

File: docs/QUICKSTART.md [hunk 1]
```diff
@@ -73,6 +77,10 @@ customresourcedefinition.apiextensions.k8s.io/users.grafana.k8spin.cloud created
 Before deploying the Grafana multi-tenant operator, create a secret with the following variables:
 
 - `GRAFANA_MULTI_TENANT_OPERATOR_HOST`: [REQUIRED] Grafana endpoint. *Example: `grafana.monitoring.svc.cluster.local:3000`*
+- `GRAFANA_MULTI_TENANT_OPERATOR_PORT`: [OPTIONAL] Grafana service port. *Defaults to `None`.*
+- `GRAFANA_MULTI_TENANT_OPERATOR_PROTOCOL`: [OPTIONAL] Scheme to use. *Defaults to `http`.*
+- `GRAFANA_MULTI_TENANT_OPERATOR_VERIFY`: [OPTIONAL] SSL Verification check. *Defaults to `True`.*
```
[the remark this answers is not in the document]
Comment (angelbarrera92) [reply docs/QUICKSTART.md:82]: Now that i reviewed carefully, I've spotted a couple of issues:

`GRAFANA_MULTI_TENANT_OPERATOR_TIMEOUT` and `GRAFANA_MULTI_TENANT_OPERATOR_PROTOCOL`:  Requires a default value as both has them in the init method.

Take this snippet as an example:

```diff
diff --git a/grafana/__init__.py b/grafana/__init__.py
index ab5c5d0..9683373 100644
--- a/grafana/__init__.py
+++ b/grafana/__init__.py
@@ -9,11 +9,11 @@ GRAFANA_MULTI_TENANT_OPERATOR_HOST = os.getenv(
 GRAFANA_MULTI_TENANT_OPERATOR_PORT = os.getenv(
     "GRAFANA_MULTI_TENANT_OPERATOR_PORT")
 GRAFANA_MULTI_TENANT_OPERATOR_PROTOCOL = os.getenv(
-    "GRAFANA_MULTI_TENANT_OPERATOR_PROTOCOL")
+    "GRAFANA_MULTI_TENANT_OPERATOR_PROTOCOL", "http")
 GRAFANA_MULTI_TENANT_OPERATOR_VERIFY = os.getenv(
     "GRAFANA_MULTI_TENANT_OPERATOR_VERIFY", True) in (True, 'true')
-GRAFANA_MULTI_TENANT_OPERATOR_TIMEOUT = os.getenv(
-"GRAFANA_MULTI_TENANT_OPERATOR_TIMEOUT")
+GRAFANA_MULTI_TENANT_OPERATOR_TIMEOUT = float(os.getenv(
+"GRAFANA_MULTI_TENANT_OPERATOR_TIMEOUT", "5.0"))
 GRAFANA_MULTI_TENANT_OPERATOR_ADMIN_USERNAME = os.getenv(
     "GRAFANA_MULTI_TENANT_OPERATOR_ADMIN_USERNAME", "admin")
 GRAFANA_MULTI_TENANT_OPERATOR_ADMIN_PASSWORD = os.getenv(
```

Give it a try. Once solved, we can continue with the merge :)

**Review by angelbarrera92 — CHANGES_REQUESTED** [review]

File: docs/QUICKSTART.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (unitygilles) [reply docs/QUICKSTART.md:82]: @angelbarrera92 thank you for the review! I added your suggestion of default http and 5.0 float values in my latest push.

Tested the new changes against an HTTPS instance and HTTP instance. 

For the HTTP instance, we here are the secret values I am passing:

```bash
$ env | grep -i grafana_multi
GRAFANA_MULTI_TENANT_OPERATOR_ADMIN_PASSWORD=admin
GRAFANA_MULTI_TENANT_OPERATOR_ADMIN_USERNAME=admin
GRAFANA_MULTI_TENANT_OPERATOR_HOST=grafana
```

Here's the output of the operator with the HTTP instance.

```bash
[2020-12-01 14:41:36,798] kopf.objects         [DEBUG   ] [grafana/example] Adding the finalizer, thus preventing the actual deletion.
[2020-12-01 14:41:36,799] kopf.objects         [DEBUG   ] [grafana/example] Patching with: {'metadata': {'finalizers': ['kopf.zalando.org/KopfFinalizerMarker']}}
[2020-12-01 14:41:36,912] kopf.objects         [DEBUG   ] [grafana/example] Creation event: {'apiVersion': 'grafana.k8spin.cloud/v1', 'kind': 'Organization', 'metadata': {'annotations': {'kubectl.kubernetes.io/last-applied-configuration': '{"apiVersion":"grafana.k8spin.cloud/v1","kind":"Organization","metadata":{"annotations":{},"name":"example","namespace":"grafana"},"spec":{}}\n'}, 'creationTimestamp': '2020-12-01T14:41:36Z', 'finalizers': ['kopf.zalando.org/KopfFinalizerMarker'], 'generation': 1, 'managedFields': [{'apiVersion': 'grafana.k8spin.cloud/v1', 'fieldsType': 'FieldsV1', 'fieldsV1': {'f:metadata': {'f:finalizers': {'.': {}, 'v:"kopf.zalando.org/KopfFinalizerMarker"': {}}}}, 'manager': 'kopf', 'operation': 'Update', 'time': '2020-12-01T14:41:36Z'}, {'apiVersion': 'grafana.k8spin.cloud/v1', 'fieldsType': 'FieldsV1', 'fieldsV1': {'f:metadata': {'f:annotations': {'.': {}, 'f:kubectl.kubernetes.io/last-applied-configuration': {}}}, 'f:spec': {}}, 'manager': 'kubectl', 'operation': 'Update', 'time': '2020-12-01T14:41:36Z'}], 'name': 'example', 'namespace': 'grafana', 'resourceVersion': '4188071', 'selfLink': '/apis/grafana.k8spin.cloud/v1/namespaces/grafana/organizations/example', 'uid': 'b73c333f-ebf9-4b2d-9fdf-a547d72607e4'}, 'spec': {}}
[2020-12-01 14:41:36,912] kopf.objects         [DEBUG   ] [grafana/example] Handler 'create_organization' is invoked.
[2020-12-01 14:41:36,914] urllib3.connectionpo [DEBUG   ] Starting new HTTP connection (1): grafana:80
[2020-12-01 14:41:36,937] urllib3.connectionpo [DEBUG   ] http://grafana:80 "POST /api/orgs HTTP/1.1" 200 44
[2020-12-01 14:41:36,958] urllib3.connectionpo [DEBUG   ] http://grafana:80 "POST /api/user/using/2 HTTP/1.1" 200 41
[2020-12-01 14:41:36,977] urllib3.connectionpo [DEBUG   ] http://grafana:80 "POST /api/user/using/1 HTTP/1.1" 200 41
[2020-12-01 14:41:36,995] urllib3.connectionpo [DEBUG   ] http://grafana:80 "POST /api/user/using/2 HTTP/1.1" 200 41
[2020-12-01 14:41:37,012] urllib3.connectionpo [DEBUG   ] http://grafana:80 "POST /api/user/using/1 HTTP/1.1" 200 41
[2020-12-01 14:41:37,013] kopf.objects         [INFO    ] [grafana/example] Handler 'create_organization' succeeded.
[2020-12-01 14:41:37,013] kopf.objects         [INFO    ] [grafana/example] Creation event is processed: 1 succeeded; 0 failed.
[2020-12-01 14:41:37,013] kopf.objects         [DEBUG   ] [grafana/example] Patching with: {'status': {'create_organization': {'message': 'Organization created', 'orgId': 2}}, 'metadata': {'annotations': {'kopf.zalando.org/last-handled-configuration': '{"spec":{}}\n'}}}
```

[the remark this answers is not in the document]
Comment (angelbarrera92) [reply docs/QUICKSTART.md:82]: Now it looks good to me!

**Review by angelbarrera92 — APPROVED** [review]
LGTM!

**Review by paurosello — APPROVED** [review]
Thanks for the contribution! Looks good

**Pull request merged**